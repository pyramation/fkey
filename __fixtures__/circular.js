export default {
  data: {
    _meta: {
      tables: [
        {
          name: 'Tree',
          foreignKeyConstraints: []
        },
        {
          name: 'Commit',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Tree'
              }
            },
            {
              refTable: {
                name: 'User'
              }
            },
            {
              refTable: {
                name: 'Ref'
              }
            }
          ]
        },
        {
          name: 'Ref',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'Commit'
              }
            }
          ]
        },
        {
          name: 'User',
          foreignKeyConstraints: [
            {
              refTable: {
                name: 'User'
              }
            }
          ]
        }
      ]
    }
  }
};
