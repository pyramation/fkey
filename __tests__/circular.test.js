import circular from '../__fixtures__/circular';
import parse from '../src';

it('circular', async () => {
  expect(parse(circular.data._meta)).toMatchSnapshot();
});
