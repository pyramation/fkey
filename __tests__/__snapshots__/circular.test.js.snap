// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circular 1`] = `
Object {
  "Commit": Object {
    "resolved": Array [
      "Tree",
      "User",
      "Ref",
      "Commit",
    ],
    "unresolved": Array [
      "Commit",
      "Tree",
      "User",
      "Ref",
    ],
  },
  "EVERYTHING_INCLUDED": Object {
    "resolved": Array [
      "Tree",
      "User",
      "Ref",
      "Commit",
      "EVERYTHING_INCLUDED",
    ],
    "unresolved": Array [
      "EVERYTHING_INCLUDED",
      "Tree",
      "Commit",
      "User",
      "Ref",
    ],
  },
  "Ref": Object {
    "resolved": Array [
      "Tree",
      "User",
      "Commit",
      "Ref",
    ],
    "unresolved": Array [
      "Ref",
      "Commit",
      "Tree",
      "User",
    ],
  },
  "Tree": Object {
    "resolved": Array [
      "Tree",
    ],
    "unresolved": Array [
      "Tree",
    ],
  },
  "User": Object {
    "resolved": Array [
      "User",
    ],
    "unresolved": Array [
      "User",
    ],
  },
}
`;

exports[`regular 1`] = `
Object {
  "Action": Object {
    "resolved": Array [
      "User",
      "Action",
    ],
    "unresolved": Array [
      "Action",
      "User",
    ],
  },
  "ActionGoal": Object {
    "resolved": Array [
      "User",
      "Action",
      "Goal",
      "ActionGoal",
    ],
    "unresolved": Array [
      "ActionGoal",
      "Action",
      "User",
      "Goal",
    ],
  },
  "ActionItem": Object {
    "resolved": Array [
      "User",
      "Action",
      "ActionItem",
    ],
    "unresolved": Array [
      "ActionItem",
      "Action",
      "User",
    ],
  },
  "ActionResult": Object {
    "resolved": Array [
      "User",
      "Action",
      "ActionResult",
    ],
    "unresolved": Array [
      "ActionResult",
      "Action",
      "User",
    ],
  },
  "EVERYTHING_INCLUDED": Object {
    "resolved": Array [
      "User",
      "Action",
      "Goal",
      "ActionGoal",
      "ActionItem",
      "ActionResult",
      "GoalExplanation",
      "NewsUpdate",
      "OrganizationProfile",
      "UserAction",
      "UserActionItem",
      "UserActionResult",
      "UserCharacteristic",
      "UserConnection",
      "UserContact",
      "UserEmail",
      "UserPassAction",
      "UserProfile",
      "UserSavedAction",
      "UserSetting",
      "ZipCode",
      "EVERYTHING_INCLUDED",
    ],
    "unresolved": Array [
      "EVERYTHING_INCLUDED",
      "ActionGoal",
      "Action",
      "User",
      "Goal",
      "ActionItem",
      "ActionResult",
      "GoalExplanation",
      "NewsUpdate",
      "OrganizationProfile",
      "UserActionItem",
      "UserAction",
      "UserActionResult",
      "UserCharacteristic",
      "UserConnection",
      "UserContact",
      "UserEmail",
      "UserPassAction",
      "UserProfile",
      "UserSavedAction",
      "UserSetting",
      "ZipCode",
    ],
  },
  "Goal": Object {
    "resolved": Array [
      "Goal",
    ],
    "unresolved": Array [
      "Goal",
    ],
  },
  "GoalExplanation": Object {
    "resolved": Array [
      "Goal",
      "GoalExplanation",
    ],
    "unresolved": Array [
      "GoalExplanation",
      "Goal",
    ],
  },
  "NewsUpdate": Object {
    "resolved": Array [
      "NewsUpdate",
    ],
    "unresolved": Array [
      "NewsUpdate",
    ],
  },
  "OrganizationProfile": Object {
    "resolved": Array [
      "User",
      "OrganizationProfile",
    ],
    "unresolved": Array [
      "OrganizationProfile",
      "User",
    ],
  },
  "User": Object {
    "resolved": Array [
      "User",
    ],
    "unresolved": Array [
      "User",
    ],
  },
  "UserAction": Object {
    "resolved": Array [
      "User",
      "Action",
      "UserAction",
    ],
    "unresolved": Array [
      "UserAction",
      "User",
      "Action",
    ],
  },
  "UserActionItem": Object {
    "resolved": Array [
      "User",
      "Action",
      "UserAction",
      "ActionItem",
      "UserActionItem",
    ],
    "unresolved": Array [
      "UserActionItem",
      "User",
      "Action",
      "UserAction",
      "ActionItem",
    ],
  },
  "UserActionResult": Object {
    "resolved": Array [
      "User",
      "Action",
      "UserAction",
      "ActionResult",
      "UserActionResult",
    ],
    "unresolved": Array [
      "UserActionResult",
      "User",
      "Action",
      "UserAction",
      "ActionResult",
    ],
  },
  "UserCharacteristic": Object {
    "resolved": Array [
      "User",
      "UserCharacteristic",
    ],
    "unresolved": Array [
      "UserCharacteristic",
      "User",
    ],
  },
  "UserConnection": Object {
    "resolved": Array [
      "User",
      "UserConnection",
    ],
    "unresolved": Array [
      "UserConnection",
      "User",
    ],
  },
  "UserContact": Object {
    "resolved": Array [
      "User",
      "UserContact",
    ],
    "unresolved": Array [
      "UserContact",
      "User",
    ],
  },
  "UserEmail": Object {
    "resolved": Array [
      "User",
      "UserEmail",
    ],
    "unresolved": Array [
      "UserEmail",
      "User",
    ],
  },
  "UserPassAction": Object {
    "resolved": Array [
      "User",
      "Action",
      "UserPassAction",
    ],
    "unresolved": Array [
      "UserPassAction",
      "User",
      "Action",
    ],
  },
  "UserProfile": Object {
    "resolved": Array [
      "User",
      "UserProfile",
    ],
    "unresolved": Array [
      "UserProfile",
      "User",
    ],
  },
  "UserSavedAction": Object {
    "resolved": Array [
      "User",
      "Action",
      "UserSavedAction",
    ],
    "unresolved": Array [
      "UserSavedAction",
      "User",
      "Action",
    ],
  },
  "UserSetting": Object {
    "resolved": Array [
      "User",
      "UserSetting",
    ],
    "unresolved": Array [
      "UserSetting",
      "User",
    ],
  },
  "ZipCode": Object {
    "resolved": Array [
      "ZipCode",
    ],
    "unresolved": Array [
      "ZipCode",
    ],
  },
}
`;
